src/terms/Hamiltonian.jl

```diff
@@ -90,8 +89,21 @@ Base.:*(H::Hamiltonian, ψ) = mul!(deepcopy(ψ), H, ψ)
     ene_ops_arr = [ene_ops(term, ψ, occ; kwargs...) for term in basis.terms]
     energies    = [eh.E for eh in ene_ops_arr]
     operators   = [eh.ops for eh in ene_ops_arr]         # operators[it][ik]
-    hks_per_k   = [[blocks[ik] for blocks in operators]  # hks_per_k[ik][it]
-                   for ik = 1:length(basis.kpoints)]
+
+    # flatten the inner arrays in case a term returns more than one operator
+    function flatten(arr)
+        ret = []
+        for a in arr
+            if a isa RealFourierOperator
+                push!(ret, a)
+            else
+                push!(ret, a...)
+            end
+        end
+        ret
+    end
```

Comment:
Why do you need this? Notice that `[1...]` and `[(1, 2)...]` does what you want.

Reply:
I need it to flatten the list (when operators return more than one term). I'm all ears if you have a better solution

Reply:
It seems it somehow should, because this is horrible. But what I came up with was no better.

Reply:
Yeah I know... haven't found anything more clever either